src/app/components/authorized/add-new-post/add-new-post.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component } from '@angular/core';
+import { GatewayService } from 'src/app/services/gateway.service';
+import { Subreddit } from 'src/types/subreddits';
+import { NewPost } from 'src/types/posts';
+import { Router } from '@angular/router';
+import { Error } from 'src/types/error';
+
+@Component({
+    selector: 'app-add-new-post',
+    templateUrl: './add-new-post.component.html',
+    styleUrls: ['./add-new-post.component.scss'],
+})
+export class AddNewPostComponent {
+    newPost: NewPost = {
+        title: '',
+        subreddit: 'Select',
+        post_type: 'text',
+        commentCount: 0,
+        posted_url: '',
+        description: '',
+    };
+
+    namesOfSubreddits: string[];
+    urlExists: boolean;
+    errorMessage$?: Error;
```

Comment:
Why are you using dollars in some properties?

Reply:
I thought that it's used in Observables, but I see now that it indicates observables, not properties made from them.. So I will delete it...